ibis/expr/tests/test_rules.py

```diff
@@ -239,3 +239,127 @@ class MyOp(ops.ValueOp):
     assert not isinstance(result.two, MyList)
     assert result.one == [MyEnum.A, MyEnum.B]
     assert result.two == [MyEnum2.B]
+
+
+def check_op_input(Op, schema, raises):
+    schema = ibis.Schema.from_tuples(schema)
+    table = ibis.table(schema)
+    if not raises:
+        assert Op(table).table.equals(table)
+    else:
+        with pytest.raises(IbisTypeError):
+            Op(table)
+
+
+@pytest.mark.parametrize(
+    ('schema', 'raises'),
+
+    [([('group', dt.int64),
+       ('value', dt.double)], False),
+
+     ([('group', dt.int64),
+       ('value', dt.timestamp)], True),
+
+     ([('group', dt.int64),
+       ('value', dt.double),
+       ('value2', dt.double)], False)])
+def test_table_with_schema(schema, raises):
```

Comment:
do you have an example of using with_coumn_subset with a rules that is NOT a column? (should raise)

Reply:
Just added one.